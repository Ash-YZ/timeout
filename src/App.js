import React, { useEffect, useState } from 'react'
import CheckBoxList from './components/CheckboxList'
import { getPeople } from './services/People'
import { getVenues } from './services/Venues'
import { dataReducer } from './reducers/DataReducer'
import './styles.scss'

const initialState = {
  peopleGoing: [],
  venuesData: []
}

function App() {

  const [people, setPeople] = useState([])
  const [venueCache, setVenueCache] = useState(null)
  const [state, dispatch] = React.useReducer(dataReducer, initialState)

  // Load people on first render
  useEffect(() => {
    getPeople()
      .then(resp => resp.json())
      .then(resp => setPeople(resp))
  }, [])

  const togglePerson = (e) => {
    dispatch({ type: 'CLEAR_VENUES_DATA' })
    if (e.target.checked) {
      dispatch({ type: 'ADD_PERSON', person: people.find(p => p.name === e.target.value) })
    } else {
      dispatch({ type: 'REMOVE_PERSON', person: e.target.value })
    }
  }

  const findRestaurants = () => {
    if (venueCache)
      dispatch({ type: 'SET_VENUE_DATA', venues: venueCache })
    else
      getVenues()
        .then(resp => resp.json())
        .then(venues => {
          dispatch({ type: 'SET_VENUE_DATA', venues })
          setVenueCache(venues)
        })
  }

  return (
    <div className="app">
      <header>
        <h1>Timeout lunch</h1>
      </header>
      {!!people.length &&
        <CheckBoxList
          title='Who&apos;s going?'
          list={people}
          groupName='peopleGoing'
          handleChange={togglePerson}
          handleSubmit={findRestaurants}
          sumbitLabel='Find restaurants' />
      }
      {!!state.venuesData.safe &&
        <>
          <h2>Places to go:</h2>
          <ul>
            {state.venuesData.safe.map((safeVenue, idx) =>
              <li key={idx}>{safeVenue.name}</li>)
            }
          </ul>
        </>
      }
      {!!state.venuesData.unsafe && !!state.venuesData.unsafe.length &&
        <>
          <h2>Places to avoid:</h2>
          <ul>
            {state.venuesData.unsafe.map((unsafeVenue, idx) =>
              <li key={idx}>{unsafeVenue.name}
                {!!unsafeVenue.noFoodFor &&
                  <ul>
                    Nothing for {unsafeVenue.noFoodFor.join(', ')} to eat
                  </ul>
                }
                {!!unsafeVenue.noDrinkFor &&
                  <ul>
                    Nothing for {unsafeVenue.noDrinkFor.join(', ')} to drink
                  </ul>
                }
              </li>)
            }
          </ul>
        </>
      }
    </div>
  )
}

export default App
