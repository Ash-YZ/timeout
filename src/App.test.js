import React from 'react'
import { render } from '@testing-library/react'
import App from './App'

const rendered = render(<App />)

test('renders title', () => {
  const title = rendered.getByText(/Timeout lunch/i)
  expect(title).toBeInTheDocument()
})

