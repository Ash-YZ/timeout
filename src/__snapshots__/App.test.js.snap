// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render correctly  1`] = `
<div
  className="app"
>
  <header>
    <h1>
      Timeout lunch
    </h1>
  </header>
</div>
`;
